### PR TITLE
Clarify the meaning of Shuffle::is_broadcast()

### PR DESCRIPTION
**Pull request by abadams**
I was poking at the Shuffle node, and checking its usage, and it seems that despite the comment, Shuffles that return true for is_broadcast are not the same as Broadcast nodes. Instead of repeating the input vector some number of times, it repeats a shuffle of the input vector. This means IRPrinter was incorrect. The other usages were fine.
    
This PR makes this clearer in the comment, and fixes IRPrinter.

### CODE REVIEW

File: src/IROperator.cpp [hunk 1]
```diff
@@ -502,7 +502,7 @@ Expr lossless_cast(Type t, Expr e) {
             Expr a = lossless_cast(t.narrow(), sub->a);
             Expr b = lossless_cast(t.narrow(), sub->b);
             if (a.defined() && b.defined()) {
-                return cast(t, a) + cast(t, b);
+                return cast(t, a) - cast(t, b);
```
Comment (steven-johnson) [inline src/IROperator.cpp:505]: uh, what

Comment (abadams) [reply src/IROperator.cpp:505]: Oops, a local change from the other PR snuck in. Will revert.

**Review by steven-johnson — APPROVED** [review]
LGTM but IMHO we'd be better off renaming this method further; having two different things both named "broadcast" is needlessly confusing

**Comment by abadams** [comment]
A Broadcast node is already capable of repeating vectors, so I actually think this shuffle pattern should be a Broadcast node wrapped around a smaller shuffle, but it's used in ways I don't entirely understand in HexagonOptimize, so it's not a trivial change. I was in the middle of trying to fix the lossless_cast bug though, so I though I'd at least fix the comment rather than get side-tracked.

**Comment by steven-johnson** [comment]
Ready to land?

**Pull request merged**